### PR TITLE
Add a way to register new presets easily

### PR DESCRIPTION
**Pull request by girardinsamuel**
A small PR to easily register new presets with Masonite 4 philosophy

- add provider for presets to allow registering new presets
- fix creating dirs if they don't exist
- add a `presets()` method in PackageProvider, to be able to register presets easily from packages. Of course we can also register a preset just by doing `app.make("presets").add(MySuperPreset())`.
- add a `--list` flag to preset command to be able to easily list available presets.

### CODE REVIEW

**Review by josephmancuso — APPROVED** [review]
This is a breaking change right?

**Comment by josephmancuso** [comment]
Since the preset provider is not in new projects or projects created with M4 this would actually remove the preset options from existing projects

**Comment by girardinsamuel** [comment]
> Since the preset provider is not in new projects or projects created with M4 this would actually remove the preset options from existing projects

@josephmancuso 
Hmm you're right ! Should we make temporarily this binding in an existing provider. and at next major release remove the binding and update project template to add the new provider ?

File: src/masonite/providers/FrameworkProvider.py [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
 
-class FrameworkProvider:
+
+class FrameworkProvider(Provider):
     def __init__(self, application):
         self.application = application
 
     def register(self):
-        pass
+        # @M5 remove this and add PresetsProvider in default project
```
Comment (girardinsamuel) [inline src/masonite/providers/FrameworkProvider.py:17]: For now I put PresetsProvider code here. @josephmancuso does this work for you or do you have a better idea ?

**Review by josephmancuso — APPROVED** [review]

**Pull request merged**